testsuite/testclient/pyuo/testclient.py

```diff
@@ -45,21 +48,51 @@ def addTodo(self,ev):
       self.todos.append(ev)
 
   def processTodos(self):
-    while True:
-      with self.todosLock:
-        if not len(self.todos):
-          return True
-        res = self.todos.popleft()
-        todo=res["todo"]
-        arg=res.get("arg",None)
-        self.log.info("got todo: {}->{}".format(todo,arg))
-        if todo=="disconnect":
-          self.client.exit()
-          return False
-        elif todo=="speech":
-          self.client.say(arg)
-        elif todo=="move":
-          self.client.move(arg)
+    with self.todosLock:
+      if not len(self.todos):
+        return True
+      todos = self.todos.copy()
+      self.todos.clear()
+    while len(todos):
+      res = todos.popleft()
+      todo=res["todo"]
```

Comment:
Maybe it's ok for now like this, but we should consider using a dictionary of todo <-> lambdas/methods for processing each command